Update progressiveText.jelly with documentation tag

duemir opened the pull request:
The `st:documentation` and `st:attribute` tags are used by [Jenkins Development Support plugin](https://github.com/jenkinsci/idea-stapler-plugin/) for autocomplete and documentation lookup.

I did this quick change in the GitHub editor as much as a conversation starter as the suggestion. There more instances like this, so there is likely more work required.  Tags documented like this get a benefit of autocompletion and documentation and deprecated attribute validation from the IntelliJ plugin. The more tags are update with unsupported documentation format the less helpful IntelliJ plugin becomes.

<!-- Comment:
A great PR typically begins with the line below.
Replace XXXXX with the numeric part of the issue ID you created in Jira.
Note that if you want your changes backported into LTS, you need to create a Jira issue. See https://www.jenkins.io/download/lts/#backporting-process for more information.
-->

See [JENKINS-XXXXX](https://issues.jenkins.io/browse/JENKINS-XXXXX).

<!-- Comment:
If the issue is not fully described in Jira, add more information here (justification, pull request links, etc.).

 * We do not require Jira issues for minor improvements.
 * Bug fixes should have a Jira issue to facilitate the backporting process.
 * Major new features should have a Jira issue.
-->

### Testing done

<!-- Comment:
Provide a clear description of how this change was tested.
At minimum this should include proof that a computer has executed the changed lines.
Ideally this should include an automated test or an explanation as to why this change has no tests.
Note that automated test coverage is less than complete, so a successful PR build does not necessarily imply that a computer has executed the changed lines.
If automated test coverage does not exist for the lines you are changing, you must describe the scenario(s) in which you manually tested the change.
For frontend changes, include screenshots of the relevant page(s) before and after the change.
For refactoring and code cleanup changes, exercise the code before and after the change and verify the behavior remains the same.
-->

### Proposed changelog entries

- human-readable text

<!-- Comment:
The changelog entry should be in the imperative mood; e.g., write "do this"/"return that" rather than "does this"/"returns that".
For examples, see: https://www.jenkins.io/changelog/

Do not include the Jira issue in the changelog entry.
Include the Jira issue in the description of the pull request so that the changelog generator can find it and include it in the generated changelog.

You may add multiple changelog entries if applicable by adding a new entry to the list, e.g.
- First changelog entry
- Second changelog entry
-->

### Proposed upgrade guidelines

N/A

<!-- Comment:
Leave the proposed upgrade guidelines in the pull request with the "N/A" value if no upgrade guidelines are needed.
The changelog generator relies on the presence of the upgrade guidelines section as part of its data extraction process.
-->

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/core-pr-reviewers.
-->

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

duemir committed:
Update progressiveText.jelly with documentation tag

The `st:documentation` and `st:attribute` tags are used by Jenkins Development Support plugin for autocomplete and documentation lookup.

timja commented on core/src/main/resources/lib/hudson/progressiveText.jelly:
```diff
@@ -35,6 +35,16 @@ THE SOFTWARE.
+    This is used to achieve the effect of "tail -f"
+    without relying on full page reload.
+    <st:attribute name="href" use="required">URL that returns text data</st:attribute>
+	  <st:attribute name="idref" use="required">ID of the HTML element in which the result is displayed</st:attribute>
```
you've got mixed tabs and spaces in here

duemir commented:
Sure, its flawed and needs to be fixed. But what about content? Does any tool rely on those JSP style comments? Or should I go forward and try and replace all of them with `st:documentation`/`st:attribute`?

timja commented:
> Sure, its flawed and needs to be fixed. But what about content? Does any tool rely on those JSP style comments? Or should I go forward and try and replace all of them with `st:documentation`/`st:attribute`?

Not that I'm aware of, content looks good and yes good to use st:documentation etc everywhere

duemir committed:
A few remaining tags with old documentation style were updated

The comment was replaced with a documentation tag. A few gaps were filled to the best of my understanding.

duemir commented:
It turned out there were only a few. I addressed the comments and updated the other two tags.

timja reviewed (approved):
Thanks!

I think Alex went through at some point and fixed most of this up.

A1exKH reviewed (approved):
@duemir this PR LGTM.

timja commented:
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

A1exKH reviewed (approved):
@duemir LGTM.
